sdk/typescript/src/keypairs/ed25519/keypair.ts

```diff
@@ -63,17 +62,26 @@ export class Ed25519Keypair extends Keypair {
 	 * This is NOT the private scalar which is result of hashing and bit clamping of
 	 * the raw secret key.
 	 *
-	 * The sui.keystore key is a list of Base64 encoded `flag || privkey`. To import
-	 * a key from sui.keystore to typescript, decode from base64 and remove the first
-	 * flag byte after checking it is indeed the Ed25519 scheme flag 0x00 (See more
-	 * on flag for signature scheme: https://github.com/MystenLabs/sui/blob/818406c5abdf7de1b80915a0519071eec3a5b1c7/crates/sui-types/src/crypto.rs#L1650):
+	 * The sui.keystore key is a list of Bech32 encoded `flag || privkey`. To import
+	 * a key from sui.keystore to typescript, decode from bech32 and parse to the 
+	 * corresponding keypair based on scheme. 
 	 * ```
-	 * import { Ed25519Keypair, fromB64 } from '@mysten/sui.js';
-	 * const raw = fromB64(t[1]);
-	 * if (raw[0] !== 0 || raw.length !== PRIVATE_KEY_SIZE + 1) {
-	 *   throw new Error('invalid key');
+	 * 	const {prefix, words} = bech32.decode(keypair.privateKey);
+	 * 	if (prefix != SUI_PRIVATE_KEY_PREFIX) {
+	 *		throw new Error('invalid key');
+	 * }
+	 * const flaggedPrivateKey = new Uint8Array(bech32.fromWords(words));
+     * const secretKey = flaggedPrivateKey.slice(1);
```

Comment:
I think we should probably accept both the old and new formats in fromSecretKey directly, I don't think we want require the bech32 decoding to be done manually by users

Reply:
the goal here is to not allow (see internal doc for reasons) at all. if someone indeed want to import a hex and confirmed it is a private key, use the conversion CLI tool.